language/move-lang/stdlib/modules/libra_account.move

```diff
@@ -52,48 +62,86 @@ module LibraAccount {
         amount: u64,
         // The address that sent the coin
         payer: address,
+        // Metadata associated with the payment
+        metadata: bytearray,
     }
 
-    // Creates a new account at `fresh_address` with the `initial_balance` deducted from the
-    // transaction sender's account
-    public fun create_new_account(fresh_address: address, initial_balance: u64) acquires T {
-        let account = T {
-            authentication_key: AddressUtil::address_to_bytes(fresh_address),
-            balance: LibraCoin::zero(),
-            delegated_key_rotation_capability: false,
-            delegated_withdrawal_capability: false,
-            received_events: Event::new_event_handle(),
-            sent_events: Event::new_event_handle(),
-            sequence_number: 0,
-        };
-        save_account(fresh_address, account);
-
-        if (initial_balance > 0) pay_from_sender(fresh_address, initial_balance);
+    /// Events
+    // A resource representing the counter used to generate uniqueness under each account. There won't be destructor for
+    // this resource to guarantee the uniqueness of the generated handle.
+    resource struct EventHandleGenerator {
+        // A monotonically increasing counter
+        counter: u64,
     }
 
-    // Save an account to a given address if the address does not have an account resource yet
-    native fun save_account(addr: address, account: Self::T);
+    // A handle for a event such that:
+    // 1. Other modules can emit event to this handle.
+    // 2. Storage can use this handle to prove the total number of events that happened in the past.
+    resource struct EventHandle<T> {
+        // Total number of events emitted to this event stream.
+        counter: u64,
+        // A globally unique ID for this event stream.
+        guid: bytearray,
+    }
 
     // Deposits the `to_deposit` coin into the `payee`'s account
     public fun deposit(payee: address, to_deposit: LibraCoin::T) acquires T {
-        // Check that the `to_deposit` coin is non-zero
-        let amount = LibraCoin::value(&to_deposit);
-        Transaction::assert(amount > 0, 7);
+        // Since we don't have bytearray literals in the source language at
```

Comment:
What's the plan for this? If we're going to add something in the language, it seems like this should have a "FIXME" comment or something. If we have no plans for that, and the u64_to_bytes is the right way to do it, then we should skip the comment here.

Reply:
I think we should definitely have these. I'll add a FIXME.